### PR TITLE
[VEN-1554][VEN-1555]: add fork tests for reward operations

### PR DESCRIPTION
**Pull request by GitGuru7**
Resolves VEN-1554, VEN-1555

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: tests/hardhat/Fork/RewardsForkTest.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+const FORK_TESTNET = process.env.FORK_TESTNET === "true";
+const FORK_MAINNET = process.env.FORK_MAINNET === "true";
+const MANTISSA_ONE = convertToUnit(1, 18);
+
+let ADMIN: string;
+let ORACLE_ADMIN: string;
```
Comment (chechu) [inline tests/hardhat/Fork/RewardsForkTest.ts:31]: We don't use it

File: tests/hardhat/Fork/RewardsForkTest.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+let acc2: string;
+let BSW: string;
+let COMPTROLLER: string;
+let VBSW: string;
+let REWARD_DISTRIBUTOR1: string;
+let REWARD_DISTRIBUTOR2: string;
+
+if (FORK_TESTNET) {
+  ADMIN = "0x2Ce1d0ffD7E869D9DF33e28552b12DdDed326706";
+  ORACLE_ADMIN = "0xce10739590001705F7FF231611ba4A48B2820327";
```
Comment (chechu) [inline tests/hardhat/Fork/RewardsForkTest.ts:43]: We don't use it

File: tests/hardhat/Fork/RewardsForkTest.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+let mintAmount: BigNumberish;
+let bswBorrowAmount: BigNumberish;
+
+async function configureTimelock() {
+  impersonatedTimelock = await initMainnetUser(ADMIN, ethers.utils.parseUnits("2"));
+}
+
+async function configureVToken(vTokenAddress: string) {
+  const VToken = VToken__factory.connect(vTokenAddress, impersonatedTimelock);
+  return VToken;
```
Comment (chechu) [inline tests/hardhat/Fork/RewardsForkTest.ts:77]:
```suggestion
  return VToken__factory.connect(vTokenAddress, impersonatedTimelock);
```

File: tests/hardhat/Fork/RewardsForkTest.ts [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  await tx.wait();
+
+  tx = await accessControlManager
+    .connect(impersonatedTimelock)
+    .giveCallPermission(comptroller.address, "setMarketBorrowCaps(address[],uint256[])", ADMIN);
+  await tx.wait();
+}
+
+if (FORK_TESTNET || FORK_MAINNET) {
+  describe("Reduce Reserves", async () => {
```
Comment (chechu) [inline tests/hardhat/Fork/RewardsForkTest.ts:95]: Rewards distributions

File: tests/hardhat/Fork/RewardsForkTest.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+      await mintVTokens(acc1Signer, bsw, vBSW, mintAmount);
+      await mine(3000000);
+      await vBSW.accrueInterest();
+
+      // Reward1 calculations for user 1
+      let supplierAccruedExpected = await computeSupplyRewards(rewardDistributor1, VBSW, vBSW, acc1);
+      let supplierAccruedCurrent = await rewardDistributor1.rewardTokenAccrued(acc1);
+      expect(supplierAccruedExpected).equals(supplierAccruedCurrent);
+
+      //   Reward2 calculations for user 1
```
Comment (chechu) [inline tests/hardhat/Fork/RewardsForkTest.ts:201]:
```suggestion
      // Reward2 calculations for user 1
```

File: tests/hardhat/Fork/RewardsForkTest.ts (hunk 1 above)
Comment (GitGuru7) [reply tests/hardhat/Fork/RewardsForkTest.ts]: Fixed [here](https://github.com/VenusProtocol/isolated-pools/pull/236/commits/3c4f565094eaf1e27db7e5a20156f7f812a427f8)

File: tests/hardhat/Fork/RewardsForkTest.ts (hunk 2 above)
Comment (GitGuru7) [reply tests/hardhat/Fork/RewardsForkTest.ts]: Fixed [here](https://github.com/VenusProtocol/isolated-pools/pull/236/commits/3c4f565094eaf1e27db7e5a20156f7f812a427f8)

File: tests/hardhat/Fork/RewardsForkTest.ts (hunk 3 above)
Comment (GitGuru7) [reply tests/hardhat/Fork/RewardsForkTest.ts]: Fixed [here](https://github.com/VenusProtocol/isolated-pools/pull/236/commits/3c4f565094eaf1e27db7e5a20156f7f812a427f8)

File: tests/hardhat/Fork/RewardsForkTest.ts (hunk 4 above)
Comment (GitGuru7) [reply tests/hardhat/Fork/RewardsForkTest.ts]: Fixed [here](https://github.com/VenusProtocol/isolated-pools/pull/236/commits/3c4f565094eaf1e27db7e5a20156f7f812a427f8)

File: tests/hardhat/Fork/RewardsForkTest.ts (hunk 5 above)
Comment (GitGuru7) [reply tests/hardhat/Fork/RewardsForkTest.ts]: Fixed [here](https://github.com/VenusProtocol/isolated-pools/pull/236/commits/3c4f565094eaf1e27db7e5a20156f7f812a427f8)

**Review by chechu — APPROVED** [review]

**Pull request merged**